wrapper/src/main/java/software/amazon/jdbc/util/events/EventPublisher.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.util.events;
+
+public interface EventPublisher {
+  /**
+   * Register the given subscriber for the given event classes.
+   *
+   * @param subscriber   the subscriber to be notified when the given event classes occur.
+   * @param eventClasses the classes of event that the subscriber should be notified of.
+   */
+  void subscribe(EventSubscriber subscriber, Class<?>... eventClasses);
```

Comment:
Class<? extends Event>

Reply:
When I make this change I get the warning "Possible heap pollution from parameterized vararg type", so maybe I'll make this a SetClass<? extends Event> instead of a vararg